packages/api/src/external/carequality/document/process-outbound-document-query-resps.ts

```diff
@@ -298,6 +252,160 @@ function buildInterrupt({
   };
 }
 
+async function getRespWithDocsToDownload({
+  cxId,
+  patientId,
+  requestId,
+  response,
+}: OutboundDocQueryRespParam): Promise<OutboundDocumentQueryResp[]> {
+  const respWithDocsToDownload: OutboundDocumentQueryResp[] = [];
+
+  await executeAsynchronously(
+    response,
+    async gwResp => {
+      const resultsWithMetriportId = await updateDocumentReferencesWithMetriportId({
+        cxId,
+        patientId,
+        requestId,
+        response: gwResp,
+      });
+      const docRefs = resultsWithMetriportId.flatMap(result => result.documentReference ?? []);
+      const docRefsWithMetriportId = filterDocRefsWithMetriportId(docRefs);
+      const docsToDownload = await getNonExistentDocRefs(docRefsWithMetriportId, patientId, cxId);
+
+      if (docsToDownload.length === 0) {
+        return;
+      }
+
+      respWithDocsToDownload.push({
+        ...gwResp,
+        documentReference: docsToDownload,
+      });
+    },
+    {
+      numberOfParallelExecutions: 20,
+    }
+  );
+
+  return respWithDocsToDownload;
+}
+
+async function updateDocumentReferencesWithMetriportId({
```

Comment:
nit: can we use another verb/action than "update" here, please? It gives the idea that we're updating the DocRefs, but we're returning new versions of it. Maybe `getDocumentReferencesWithMetriportId()`?